### DISCUSSION TITLE
Shutter Speed, Bracketing, Lightroom HDR, Level in vertical

### DISCUSSION THREAD

**Discussion by gregorpesek** [gopro/labs#167] [Feature Requests]
Hi. These are my requests for future updates 🙏

1. Shutter Speed Control
  Because Shutter Angle doesn't work in photo mode!

2. Complete Shutter Speed Control no matter if NTSC or PAL format: (1/24, 1/48, 1/50, 1/60, 1/100, 1/ 120, 1/200, 1/240, etc.)
  Why? Anti-Flicker issue: I live in a country with 50hz electricity (PAL). But I prefer to record everything in NTSC (30, 60,... FPS). So right now when shooting inside with flickering lights I'm forced to shoot in PAL, but all of my footage from outside is NTSC. And when I combine the clips into an edit either to NTSC or PAL frame rate; the one that isn't  in the correct format, will be stuttering because of missing frames. 

   FIX: The only way to solve my problem is if I had the option to record at 30/60 FPS with Shutter speed at 1/50, 1/100, etc. to match the frequency of the flickering lights.

3. HDR photos/Bracketing (Crucial for GoPro MAX)
  For Better dynamic range. Right now you can use !10Nx-2ti1M1!Px0ti1M1!Px1ti1M1!P command which takes consecutive pictures at different exposure levels. Problem with this is, the process is very slow. For 3 pictures from first to last it takes about 7 seconds to complete the process. With this it becomes a problem when there is wind present that causes things to move. Branches for example. Mirrorless/DSLR cameras for example take no more than a second for the same process. It would be great if the process would take less time to complete.

4. HDR merge in Lightroom (Crucial for GoPro MAX)
 It would be great if we would be able to merge the Photos in Lightroom with their HDR feature for way faster workflow. 

5. Make the level work when shooting in vertical aswell. And also make it available for 10 please 🙏

Cheers!

**Comment by andrerav** [comment]
> 2. Complete Shutter Speed Control

I think you can use the EXPN and EXPX settings for this? Edit: If you have a GoPro 11, that is :)

**Reply by gregorpesek** [reply]
Unfortunately cannot. I tried it and all what it does when setting to 1/50 is it changes it to 1/60 automatically 

Edit:
All what it does when setting to 1/50, it changes it to 1/60 automatically

**Reply by dnewman-gpsw** [reply]
Re 2.  No, it works correctly.  I don't know what this is "setting to 1/50 is it changes it to 1/60 automatically", maybe you are referring to the UI. The EXPX and EXPN have nothing to do with UI, they are deeper controls than that.  If you set both EXPN and EXPX to 50, and leave the shutter control in the UI to automatic, the camera will run the shutter at 1/50.  
![image](https://user-images.githubusercontent.com/10856084/208313708-f9cfc874-debc-4004-ac16-189bdb0b1587.png)

**Reply by gregorpesek** [reply]
Lovely! It does work. Like you said I just had to put shutter on auto that I didn't know. However this command still doesn't fixes the issue where you want automatic shutter speed without flickering¹ and also doesn't work for photo mode. Could you make the command work for photo mode as well? 

¹ Think what gopro should do from the top is allow the user to change any frame rate (50, 60, etc.) in the frame rates setting and the anti - flicker should only control the shutter frequency. I know it might seem counter intuitive for normal use, but it's either that or a stuttering video... Which would you prefer? 

_____________________________________________
 How about my other requests?

**Reply by dnewman-gpsw** [reply]
MAX is not likely to see these updates (one seems to be an Adobe function, not a camera request.)  Labs is mostly a command and control extension, rarely does it alter the fundamentals of camera functionality. For the exposure control hacks, it uses existing features, often just setting different limits, or turning off some features. A complete overhaul of camera exposure rules is not a Labs hack, that is core team work, so beyond the scope of Labs.  It is a good idea, just not for Labs.

For auto exposure to support anti-flicker, either shoot an fps at the line frequency, or fix the shutter (EXPN/EXPX=x*line  ), and allow the ISO to gain up for the indoor elements (and denoise in post.)  I think shooting at the line frequency would be easier.

**Reply by gregorpesek** [reply]
I'm confused what x*line means. Could you write down the exact command you gnk might fit me?
How about the level when holding the camera vertical and is it something that the 10 could get as well?

**Comment by dnewman-gpsw** [comment]
Level could fixed, yes. Too many request is one post, I missed it.

x*line was an over simplification. Use these guides:

Under 50Hz lightning
fps - Safe shutter speed
60p - 1/100	
30p - 1/33.3, 1/50, 1/100	
24p - 1/33.3, 1/50, 1/100

Under 60Hz lightning
fps - Safe shutter speed
50p - 1/60, 1/120	
25p - 1/40, 1/60, 1/120
24p - any

From https://www.urbanvideo.ca/avoid-video-flicker and https://www.instyleled.co.uk/avoiding-led-flicker-pal-vs-ntsc/

**Reply by gregorpesek** [reply]
https://user-images.githubusercontent.com/45019178/209597047-af99e5bc-0ee9-4852-af95-55af7439d1d9.mp4

Well finally when it came to a shooting day I was presented with another problem. The EXPN and EXPM only work for 50 and 100, so 1/50 and 1/100 shutter speed. Yesterday I wanted to shoot at 1/200 because I had enough light to do slow-motion but it still flickered, same with 400 and 800. So I changed back to PAL and the flicker was gone no matter which shutter speed I choose. It would be great if you could make it work for all shutter speeds not just 50 and 100

Edit: i posted a video recording the gym with my phone that has no flicker showing the preview of the gopro with visible flicker

**Reply by dnewman-gpsw** [reply]
No other shutter speeds can work when mixing these frame rates and power line frequency.  Nothing Labs or any camera can do to fix that
 https://www.red.com/red-101/flicker-free-video-tutorial

**Reply by gregorpesek** [reply]
Thank you, I didn't know that. I'm wondering how it's not working for 1/300, since both 60 and 50 are factor of 300..

**Reply by dnewman-gpsw** [reply]
Not how it works.  The above link explains it.  Basically you are sampling different parts of a sine wave, you want the sampled regions to match in power.
